components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultLogoutRequestHandler.java

```diff
@@ -314,8 +314,6 @@ protected void sendResponse(HttpServletRequest request, HttpServletResponse resp
             String sessionDataKeyParam = FrameworkConstants.SESSION_DATA_KEY + "=" +
                     URLEncoder.encode(context.getCallerSessionKey(), "UTF-8");
             redirectURL = FrameworkUtils.appendQueryParamsStringToUrl(redirectURL, sessionDataKeyParam);
-        } else {
-            redirectURL = context.getCallerPath();
         }
```

Comment:
Is this intentional?

Reply:
Yes, if the else block is there it overrides the redirectURL set in [line 287](https://github.com/wso2/carbon-identity-framework/pull/4372/files#diff-09fa487817b3a0de64491dd4d4573be7fd881e1662fced5fa35db6ee89e8be6bR287) if the if condition in line 294 fails.

Reply:
Yes, but can we assure line 294 is always true. Trying to figure out whether this else block was unnecessary before your changes. If so it's fine, otherwise line 336 would redirect null value.